RHDEVDOCS-3279 Integrate Shared Resources Operator with Cluster Storage (#41898)
openshift/openshift-docs: modules/ephemeral-storage-integration-between-shared-resources-insights-operator-and-openshift-builds.adoc

:_content-type: REFERENCE

[id="ephemeral-storage-integration-between-shared-resources-insights-operator-and-openshift-builds_{context}"]

> = Integration between shared resources, Insights Operator, and OpenShift Builds

Reviewer: ```suggestion
= Integration between Shared Resources, Insights Operator, and {product-title} Builds
```

Author: Titles and "shared resources" use sentence case. Updated  "OpenShift Builds" to "{product-title} Builds." Thanks.

[role="_abstract"]

> Integration between shared resources, Insights Operator, and OpenShift Builds makes using Red Hat subscriptions (RHEL entitlements) easier.

Reviewer: ```suggestion
Integration between Shared Resources, Insights Operator, and OpenShift Builds makes using Red Hat subscriptions (RHEL entitlements) easier in {product-title} Builds.
```

Author: The term, "shared resources," uses sentence case throughout the OpenShift docs. I've updated "OpenShift Builds" to "{product-title} Builds." Thanks.

> Previously, in {product-title} 4.9.x and earlier, you manually imported your credentials or copied them to each project or namespace where you were running builds.

Reviewer: create a link to the existing Builds documentation for this, like in 4.9, it is https://docs.openshift.com/container-platform/4.9/cicd/builds/running-entitled-builds.html

Author: Addressed in another related comment for this file.

> Previously, in {product-title} 4.9.x and earlier, you manually imported your credentials or copied them to each project or namespace where you were running builds.

Reviewer: ```suggestion
Previously, in {product-title} 4.9.x and earlier, you manually imported your credentials and copied them to each project or namespace where you were running builds.
```

Author: Updated. Thanks.

> Now, in {product-title} 4.10 and later, OpenShift Builds can employ Red Hat subscriptions (RHEL entitlements) by using shared resources and the simple content access feature provided by Insights Operator:

Reviewer: ```suggestion
Now, in {product-title} 4.10 and later, OpenShift Builds can employ Red Hat subscriptions (RHEL entitlements) by using Shared Resources and the Simple Content Access feature provided by Insights Operator:
```

Reviewer: also, add a link to the 4.10 version of https://docs.openshift.com/container-platform/4.9/support/remote_health_monitoring/insights-operator-simple-access.html

Author: The terms "shared resources" and "simple content access" are sentence case throughout the OpenShift docs.

Author: Added the following cross-references to the `storage/container_storage_interface/ephemeral-storage-shared-resource-csi-driver-operator.adoc` file:
```
.Additional resources

xref:../../support/remote_health_monitoring/insights-operator-simple-access.adoc#insights-operator-simple-access[Importing simple content access certificates with Insights Operator]
xref:../../cicd/builds/running-entitled-builds.html#builds-source-secrets-entitlements_running-entitled-builds[Adding subscription entitlements as a build secret]
```
To readers, these cross-references look like they're at the bottom of the `modules/ephemeral-storage-integration-between-shared-resources-insights-operator-and-openshift-builds.adoc` file.

I also added "See the links in the following "Additional resources" section." to one of the bullets in the  the `modules/ephemeral-storage-integration-between-shared-resources-insights-operator-and-openshift-builds.adoc` file.

All of this is because of a mind-boggling limitation that prevents us from using xrefs in module files.

* The simple content access feature imports your subscription credentials to a well-known `Secret` object.

> * The cluster administrator creates a `SharedSecret` custom resource (CR) around that `Secret` object and grants permission to particular projects or namespaces.

Reviewer: ```suggestion
* The cluster administrator creates a `SharedSecret` custom resource (CR) around that `Secret` object and grants permission to particular projects or namespaces.  Particularly, the 'builder' ServiceAccount must be given permission to 'use' the SharedSecret.
```

Author: Updated. Thanks.

> * Builds that run within those projects or namespaces have access to the `SharedSecret` CR and its entitled RHEL content.

Reviewer: ```suggestion
* Builds that run within those projects or namespaces can mount a CSI Volume referencing the `SharedSecret` instance and its entitled RHEL content.
```

Reviewer: Also, I would expect the example yaml for this this like we have in the blog post to be part of your updates (assuming a separate PR) to the existing Build article and using Red Hat subscriptoins.  But you will update this article at that time to link to that new content.

Author: Updated. Thanks. Keeping this comment open for now as a reminder.

:_content-type: REFERENCE

[id="ephemeral-storage-integration-between-shared-resources-insights-operator-and-openshift-builds_{context}"]

> = Integration between shared resources, Insights Operator, and OpenShift Builds

Author: @gabemontero @prietyc123 @bburt-rh New topic. Please review and approve.
